packages/ui/Scrollable/Scrollable.js

```diff
@@ -64,710 +64,499 @@ const TouchableDiv = ForwardRef({prop: 'ref'}, Touchable('div'));
-		if (this.scrolling) {
-			this.forwardScrollEvent('onScrollStop', this.getReachedEdgeInfo());
-		}
-		this.scrollStopJob.stop();
+	function clampScrollPosition () {
+		const bounds = getScrollBounds();
 
-		if (this.animator.isAnimating()) {
-			this.animator.stop();
+		if (variables.current.scrollTop > bounds.maxTop) {
+			variables.current.scrollTop = bounds.maxTop;
 		}
 
-		this.removeEventListeners();
-	}
-
-	handleResize (ev) {
-		if (ev.action === 'invalidateBounds') {
-			this.enqueueForceUpdate();
+		if (variables.current.scrollLeft > bounds.maxLeft) {
+			variables.current.scrollLeft = bounds.maxLeft;
 		}
 	}
 
-	handleResizeWindow = () => {
-		// `handleSize` in `ui/resolution.ResolutionDecorator` should be executed first.
-		setTimeout(() => {
-			const {handleResizeWindow} = this.props;
-
-			if (handleResizeWindow) {
-				handleResizeWindow();
-			}
-			this.scrollTo({position: {x: 0, y: 0}, animate: false});
-
-			this.enqueueForceUpdate();
-		});
-	}
+	// drag/flick event handlers
 
-	// TODO: consider replacing forceUpdate() by storing bounds in state rather than a non-
-	// state member.
-	enqueueForceUpdate () {
-		this.childRefCurrent.calculateMetrics(this.childRefCurrent.props);
-		this.forceUpdate();
+	function getRtlX (x) {
+		return (props.rtl ? -x : x);
 	}
 
-	clampScrollPosition () {
-		const bounds = this.getScrollBounds();
-
-		if (this.scrollTop > bounds.maxTop) {
-			this.scrollTop = bounds.maxTop;
+	const stop = useCallback(() => {
+		variables.current.animator.stop();
+		variables.current.lastInputType = null;
+		variables.current.isScrollAnimationTargetAccumulated = false;
+		startHidingThumb();
+		if (variables.current.overscrollEnabled && !variables.current.isDragging) { // not check props.overscrollEffectOn for safety
+			clearAllOverscrollEffects();
 		}
-
-		if (this.scrollLeft > bounds.maxLeft) {
-			this.scrollLeft = bounds.maxLeft;
+		if (props.stop) {
+			props.stop();
 		}
-	}
-
-	// constants
-	pixelPerLine = 39
-	scrollWheelMultiplierForDeltaPixel = 1.5 // The ratio of wheel 'delta' units to pixels scrolled.
-
-	// status
-	deferScrollTo = true
-	isScrollAnimationTargetAccumulated = false
-	isUpdatedScrollThumb = false
-
-	// overscroll
-	lastInputType = null
-	overscrollEnabled = false
-	overscrollStatus = {
-		horizontal: {
-			before: {type: overscrollTypeNone, ratio: 0},
-			after: {type: overscrollTypeNone, ratio: 0}
-		},
-		vertical: {
-			before: {type: overscrollTypeNone, ratio: 0},
-			after: {type: overscrollTypeNone, ratio: 0}
+		if (variables.current.scrolling) {
+			variables.current.scrollStopJob.start();
 		}
-	}
-
-	// bounds info
-	bounds = {
-		clientWidth: 0,
-		clientHeight: 0,
-		scrollWidth: 0,
-		scrollHeight: 0,
-		maxTop: 0,
-		maxLeft: 0
-	}
+	});
 
-	// wheel/drag/flick info
-	wheelDirection = 0
-	isDragging = false
+	const onMouseDown = useCallback((ev) => {
+		forwardWithPrevent('onMouseDown', ev, props);
```

Comment:
This callback is converted from `handle` chains, so you must check the return value from `forwardWithPrevent` to determine the execution flow should be stopped or not.

Reply:
Good point. I fixed it.